packages/contracts/src/PoolMetadataRegistry.sol

```diff
@@ -32,21 +42,27 @@ contract PoolMetadataRegistry is IPoolMetadataRegistry {
         }
     }
 
-    /// @notice Check if the caller is the pool owner
-    /// @param poolId The pool ID where the ownership will be tested.
-    /// @return bool Returns TRUE if the caller is the Pool owner, FALSE otherwise.
-    function _isPoolOwner(bytes32 poolId) public view returns (bool) {
-        (address pool,) = _vault.getPool(poolId);
+    /// @notice Returns the authorizer contract for managing access control
+    /// @return The authorizer contract
+    function _getAuthorizer() internal view override returns (IAuthorizer) {
+        return _vault.getAuthorizer();
+    }
 
-        return BasePoolAuthorization(pool).getOwner() == msg.sender ? true : false;
+    /// @notice Returns the owner of the pool that's currently being operated on
+    /// @return The owner of the pool
+    function _getOwner() internal view override returns (address) {
+        (address pool,) = _vault.getPool(_poolId);
+        return BasePoolAuthorization(pool).getOwner();
     }
 
     /// @notice Updates the pool metadata CID
     /// @param poolId The pool ID to update the metadata
     /// @param metadataCID The metadataCID related to the new pool metadata
-    function setPoolMetadata(bytes32 poolId, string memory metadataCID) public {
-        require(_isPoolRegistered(poolId), "Pool not registered");
-        require(_isPoolOwner(poolId), "Caller is not the owner");
+    function setPoolMetadata(bytes32 poolId, string memory metadataCID)
+        public
+        onlyRegisteredPool(poolId)
+        authenticate
```

Comment:
I assume this modifier is defined in `Authentication` and uses your `_getOwner` function, right? Since there's no way to know that wo knowing the context can you add a brief explanation of how that works here?